components/server/src/api/server.ts

```diff
@@ -174,9 +180,25 @@ export class API {
 
                     const context = args[1] as HandlerContext;
 
+                    const rateLimit = async (userId: string) => {
+                        try {
+                            const key = `${grpc_service}.${grpc_method}`;
+                            const options = self.config.rateLimits[key] || RateLimited.getOptions(target, prop);
+                            await self.getRateLimitter(options).consume(`${userId}_${key}`);
```

Comment:
@svenefftinge here we pass the actual key from which points should be subtracted

Reply:
yeah, took me some time to understand that it contains multiple buckets